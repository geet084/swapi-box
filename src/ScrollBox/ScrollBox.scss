.scroll {
  background-color: black;
  opacity: .7;
  color: yellow;
  height: 100px;
  padding: 0 30%;
  line-height: 2;
  perspective: 400%;
}
