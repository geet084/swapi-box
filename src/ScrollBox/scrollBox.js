import React from 'react'
import './ScrollBox.scss'

const scrollBox = ({ movie }) => {
  return (
    <marquee
      className="scroll"
      behavior="scroll"
      direction="up"
      scrollamount="1.5">
      {movie}
    </marquee>
  )
}

export default scrollBox;