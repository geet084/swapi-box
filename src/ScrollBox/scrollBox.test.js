import React from 'react'
import { shallow } from 'enzyme'
import ScrollBox from './ScrollBox'

const mockMovie = 'Some text to display here';

describe('ScrollBox', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <ScrollBox crawlText={mockMovie} />
    )
  })
  
  it('should properly render snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should receive correct props', () => {
    expect(wrapper.props().children).toEqual(mockMovie)
  })

})