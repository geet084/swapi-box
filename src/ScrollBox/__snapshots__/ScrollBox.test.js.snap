// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollBox should properly render snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <scrollBox
    crawlText="Some text to display here"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "behavior": "scroll",
      "children": <div>
        <p>
          Some text to display here
        </p>
        <p
          className="title"
        >
          <span>
                  
          </span>
        </p>
      </div>,
      "className": "scroll",
      "direction": "up",
      "scrollamount": "1.5",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            Some text to display here
          </p>,
          <p
            className="title"
          >
            <span>
                    
            </span>
          </p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Some text to display here",
          },
          "ref": null,
          "rendered": "Some text to display here",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span>
                    
            </span>,
            "className": "title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                undefined,
                "      ",
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              undefined,
              "      ",
              undefined,
            ],
            "type": "span",
          },
          "type": "p",
        },
      ],
      "type": "div",
    },
    "type": "marquee",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "behavior": "scroll",
        "children": <div>
          <p>
            Some text to display here
          </p>
          <p
            className="title"
          >
            <span>
                    
            </span>
          </p>
        </div>,
        "className": "scroll",
        "direction": "up",
        "scrollamount": "1.5",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              Some text to display here
            </p>,
            <p
              className="title"
            >
              <span>
                      
              </span>
            </p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Some text to display here",
            },
            "ref": null,
            "rendered": "Some text to display here",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span>
                      
              </span>,
              "className": "title",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  undefined,
                  "      ",
                  undefined,
                ],
              },
              "ref": null,
              "rendered": Array [
                undefined,
                "      ",
                undefined,
              ],
              "type": "span",
            },
            "type": "p",
          },
        ],
        "type": "div",
      },
      "type": "marquee",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
